feat(sounds): disabling the sounds when running cypress

DavidLuterancik opened the pull request:
This PR resolves the issue #753 

Disabled the sounds when running cypress tests with additional refactoring to avoid further redundancy on sound files played using Howl.

Please let me know if anything is wrong :)

kantord commented on workspaces/web/src/media/sound.js:
```diff
@@ -1,15 +1,17 @@
+    }
+}
+
+export default sound
```
Can you pls add a newline at the end of the file?

kantord commented on workspaces/web/src/media/sound.js:
```diff
@@ -1,15 +1,17 @@
-    fanfare: new Howl({
-        src: ["sound/fanfare.mp3"],
-    }),
+    correct: () => playAudio("sound","correct"),
```
One danger that I see here is that this way the browser will only start downloading the sound file once it should be actually played. That could lead to some extra latency when it's first played, especially on slow networks.

Before this the download of the sound file started (afaik) when the module was imported (i. e. when `new Howl` was called

kantord commented:
Hi, @DavidLuterancik, thank you!!

I think it's almost perfect, just please resolve the 2 comments that I pointed out

DavidLuterancik commented:
After testing the latency with simulated slower network, I've reverted to the previous solution with sounds. As you stated, the sounds are downloaded at the module import.

Also, added the new line at the end of the file.

kantord commented on workspaces/web/src/media/sound.js:
```diff
@@ -12,4 +12,12 @@ const sound = {
 }
 
+export const playAudio = (type,filename) => {
+    if(!window.Cypress){
```
ok, this is perfect, because this never had a preloading system before anyway. Perhaps, we can create a task for building the preloading system, but it's not relevant here! :+1:

kantord commented:
Based on the code now it seems to me though, that the "sounds effects" are not disabled at all, only the voices :thinking:  Look for other usages of howler.

DavidLuterancik commented:
I've tried to find a different solution, this seems to be working fine. I am not quite happy with the way it achieves it. Should I find a way to stop/mute only certain sounds or is this enough for running tests.

![image](https://user-images.githubusercontent.com/72352073/95109637-24fea600-073d-11eb-842e-07f93879f229.png)

kantord commented:
> I've tried to find a different solution, this seems to be working fine. I am not quite happy with the way it achieves it. Should I find a way to stop/mute only certain sounds or is this enough for running tests.
> 
> ![image](https://user-images.githubusercontent.com/72352073/95109637-24fea600-073d-11eb-842e-07f93879f229.png)

I think this is a good way!

Another way would be to wrap howler sth like this:

```
const loadSound = ({ src }) => {
  if (window.isCypress) {
     return { play: () => null }
  }
 
  return new Howler({ "src": [src] })
}

this way it will be possible to `.play()` it every time, but in cypress it will do nothing.
```

DavidLuterancik commented:
Couldn't put my finger on it with the wrap :/ hope this will do for now

kantord commented:
Yes! Looks great to me! :100:  Thank you!

kantord commented:
@allcontributors please add @DavidLuterancik for code
